### PR TITLE
InCanvasSearch: horizontal scroll bars

### PR DESCRIPTION
**Pull request by aosyatnik**
### Purpose

Link to YouTrack:
[MAGN-7645](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7645) Horizontal scroll bars are difficult to use - take out method signature so users don't have to scroll as much

Result:
image

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] The level of testing this PR includes is appropriate

### Reviewers

@Benglin 
@pboyer 

### FYIs

@lillismith

### CODE REVIEW

File: src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml [hunk 1]
```diff
@@ -130,7 +130,7 @@
                                       Style="{StaticResource InCanvasScrollViewerStyle}"
                                       CanContentScroll="True"
                                       Width="250">
-                            <ContentControl Template="{StaticResource MemberControlTemplate}" />
+                            <ContentControl Template="{StaticResource InCanvasMemberControlTemplate}" />
```
Comment (Benglin) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml]: It seems like right now we have both `MemberControlTemplate` and `InCanvasMemberControlTemplate` templates, is there a reason we don't apply this style to both the regular library search (in addition to the in-canvas search)? I'd be glad if we don't add another new template and keep our `XAML` simpler :)

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 2]
```diff
@@ -116,6 +117,26 @@ public string Name
...
+        public string UserFriendlyName
+        {
+            get
+            {
+                if (!String.IsNullOrWhiteSpace(userFriendlyName))
+                    return userFriendlyName;
+                else
+                    return name;
+            }
+            set
```
Comment (Benglin) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs]: This should be `protected set` if that is possible, I don't see how/why external codes need to set this value.

**Comment by aosyatnik** [comment]
Hi, @Benglin . I've started removing public setters from `NodeSearchElement` and noticed, that it also implements `INotifyPropertyChanged`, which is wrong. Since `NodeSearchElement` is model in fact it shouldn't notify UI somehow. So, I removed `INotifyPropertyChanged` and everything, that is connected with it.

I tried run tests, there is no fallen test.

Can I assign task [MAGN-5716](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-5716) NodeSearchElement.IsVisibleInSearch property should be removed to myself?

File: src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml (hunk 1 above)
Comment (aosyatnik) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml]: There is one difference between them. `MemberControlTemplate` uses full name with signature.
And `InCanvasMemberControlTemplate` uses user friendly name, i.e. without signature.

Comment (Benglin) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml]: Yep that's exactly my point, @aosyatnik. Let's see if @lillismith has any view on this in terms of design... Lilli, here's what we're discussing here -- there are two scenarios in which a class member method is displayed: one in regular library view when user expands a given class, and another in search results (both library and in-canvas cases). In regular library view, a class member method includes its signature (e.g. `ByCoordinates(double x, double y)`) and in search view, a class member method does not include its signature (e.g. `ByCoordinates`).

Would you think we should align these two views to only show method names and not the full signatures? (If we decide to unify those two, we can do it as separate PR, not this one.)

File: src/DynamoCore/Search/SearchElements/NodeSearchElement.cs [hunk 3]
```diff
@@ -13,14 +13,13 @@ namespace Dynamo.Search.SearchElements
     /// <summary>
     ///     Base class for all Dynamo Node search elements.
     /// </summary>
-    public abstract class NodeSearchElement : INotifyPropertyChanged, ISearchEntry, ISource<NodeModel>
+    public abstract class NodeSearchElement : ISearchEntry, ISource<NodeModel>
```
Comment (Benglin) [inline src/DynamoCore/Search/SearchElements/NodeSearchElement.cs]: Just to make sure this removal is safe, you need to confirm that these properties are only ever set once (during construction) and they do not change subsequently. If they do change (e.g. renaming the `name` or `description` of a custom node), the view needs to be updated.

**Comment by Benglin** [comment]
Sure @aosyatnik, you can take `MAGN-5716` anytime. Thanks.

File: src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml (hunk 1 above)
Comment (lillismith) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml]: I discussed with Zach and we are ok with library, browse, search, and in-canvas search showing just the method name and not the full signature.  The original intent was to just take the full signature off for in-canvas search, but we think it is ok to take it off in both places. This information is more legible in the tool tip.

Comment (Benglin) [inline src/DynamoCoreWpf/Controls/IncanvasSearchControl.xaml]: Thanks @lillismith. Hi @aosyatnik, I have created this task for you to track internally:

[MAGN-7780](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7780) Remove full method signature from library UI

That can be done as a separate PR and get this merged in first (once you've confirmed the testing I mentioned below is good).

**Comment by aosyatnik** [comment]
Hi, @Benglin .
I checked all properties, which used `INotifyPropertyChanged`. And I found, that just one is really needed to be updated in View. It's `IsVisibleInSearch`. `IsVisibleInSearch` is used for Input and Output. These nodes can be created just if current workspace is custom node workspace. So, we can't give user opportunity to create them in home workspace. I've created Action `VisibilityChanged`, it's used to notify VM, that visibility of node changed, i.e. workspace changed.

All other properties don't require View notification.

**Comment by Benglin** [comment]
Hi @aosyatnik, thanks for the notification. There's a large workspace change that is going to get in, let's wait for that before merging this in (the workspace changes are large and require the build to be in clean state before it can proceed). Sorry for the delay!

**Pull request merged**